docs/dev/new_check_howto.md

```diff
@@ -266,57 +266,106 @@ Configuration blocks in either section take the following form:
 
 ```yaml
 ## @<COMMAND> [- <ARGS>]
-## <DESCRIPTION>
+## <DESCRIPTION LINE 1>
+## <DESCRIPTION LINE 2>
 #
 <KEY>: <VALUE>
 ```
 
 Configuration blocks follow a few guidelines:
 
+* Description should not be empty
 * Placeholders should always follow this format: `<THIS_IS_A_PLACEHOLDER>`, as per the documentation [contributing guidelines][9]:
 * All required parameters are **not** commented by default.
 * All optional parameters are commented by default.
 * If a placeholder has a default value for an integration (like the status endpoint of an integration), it can be used instead of a generic placeholder.
 
 #### @param specification
 
-Practically speaking, the only command is `@param`, which is used to describe configuration blocks—primarily for documentation purposes. `@param` is implemented using this form:
+Practically speaking, the only command is `@param`, which is used to describe configuration blocks—primarily for documentation purposes. `@param` is implemented using one of the following forms:
 
 ```
-@param <name> - <type> - <required> - default:<defval>
+@param <name> - <type> - required
+@param <name> - <type> - optional - default: <defval>
 ```
 
 Arguments:
 
 * `name`: the name of the parameter, e.g. `search_string` (mandatory).
 * `type`: the data type for the parameter value (mandatory). Possible values:
+  * *boolean*
+  * *string*
   * *integer*
   * *double*
-  * *string*
-  * comma separated list comprising **one** of those three types
-* `required`: whether the parameter is required or not (mandatory). Possible values:
-    * *required*
-    * *optional*
+  * *float*
+  * *dictionary*
+  * *list\**
+  * *object*
 * `defval`: default value for the parameter; can be empty (optional).
 
+`list` and `object` variables will span over multiple lines and have special rules.
+
+* In a `list`, individual elements are not documented with the @param specification
+* In an `object` individual elements are documented. You can though ignore documentation of individual elements if you consider the object is "simple enough", in that case do not document any individual element.
```

Comment:
I would not mention to not document something.

Reply:
I updated the sentence, let me know what you think

Reply:
Great !! I actually misunderstood the sentence before, but now it's clear 👍